File: Lv.1/나누어 떨어지는 숫자 배열.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr, int divisor) {
    vector<int> answer;
    
    for(int temp: arr)
    {
        if(temp%divisor==0)
        {
            answer.push_back(temp);
        }
    }
    
    if(answer.size()>=1)
    {
      sort(answer.begin(),answer.end());
      
    }
    
    else
    {
        answer.push_back(-1);
    }
    
    return answer;
}

File: Java/LV1/나누어 떨어지는 숫자 배열.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        int[] answer={};
        
        int count = 0;
        for(int i=0;i<arr.length;i++){
            if(arr[i]%divisor==0){
                count++;
            }
        }
        
        answer = new int[count];
        
        int idx=0;
        for(int i=0;i<arr.length;i++){
            if(arr[i]%divisor==0){
                answer[idx++]=arr[i];
            }
        }
        
        if(answer.length<1){
            answer = new int[1];
            answer[0]=-1;
        }
        Arrays.sort(answer);
        return answer;
    }
}
